### ISSUE TITLE
Error with makeStore (from Data.Store.TH)

### ISSUE THREAD

**Issue by cdisselkoen** [fpco/store#139]
Let me start by saying that I'm not sure if this is actually a bug in `store` or in something else; or it may be out of scope.  However, it does impact the ability to use `store`, so any suggestions for workarounds are welcome.

For certain data types, using `makeStore` on that data type results in the error
```
    A 115-tuple is too large for GHC
      (max size is 62)
      Workaround: use nested tuples or define a data type
```

Here's a simple example reproducing the error (using the llvm-hs-pure library):
```
{-# LANGUAGE TemplateHaskell #-}

import Data.Store.TH
import LLVM.AST.Constant

makeStore ''Constant
```

Note, the same error occurs with `TH.Derive`:
```
{-# LANGUAGE TemplateHaskell #-}

import Data.Store
import LLVM.AST.Constant
import TH.Derive

$($(derive [d| instance Deriving (Store Constant) |]))
```

**Comment by cdisselkoen** [comment]
Here is [output of compiling the first example with GHC's `-ddump-splices` command](https://github.com/fpco/store/files/3087848/Lib.dump-splices.txt).  We can clearly see that there are 115-tuples being used.  Again, not sure if this is `store`'s problem or someone else's.

**Comment by mgsloan** [comment]
Ah yes, I encountered this myself, and fixed it in https://github.com/fpco/store/commit/287e252516b1a6a7b19c25dc5a584116dbf5fd4f .  I just forgot to make a release - I've just now released version `0.5.1.0`.  Please verify that this version fixes the issue for you.  Thanks!

**Comment by mgsloan** [comment]
Going to go ahead and close this, assuming that the problem has been fixed by the release.
